projects/storefrontapp-e2e-cypress/cypress/e2e/regression/product_details/show-real-time-stock-in-PDP.e2e-flaky.cy.ts

```diff
@@ -0,0 +1,146 @@
+
+export const GET_PRODUCT_AVAILABILITY_ENDPOINT_ALIAS = 'getProductDetails';
+
+export function interceptProductAvailability(productCode: string) {
+  cy.intercept(
+    'GET',
+    `${Cypress.env('OCC_PREFIX')}/apparel-uk-spa/productAvailabilities?filters=${productCode}:ST`
+  ).as(GET_PRODUCT_AVAILABILITY_ENDPOINT_ALIAS);
+
+  return GET_PRODUCT_AVAILABILITY_ENDPOINT_ALIAS;
+}
+
+export function configureInventoryDisplay(enable: boolean) {
+  cy.cxConfig({
+    cmsComponents: {
+      ProductAddToCartComponent: {
+        data: {
+          inventoryDisplay: enable,
+        },
+      },
+    },
+    features: {
+      realTimeStockDispaly: enable,
+    },
+  });
+}
+
+export function assertInventoryDisplay(
+  productCode: string,
+  alias: string,
+  functionality: string
+) {
+  cy.get(`${alias}`).then((xhr) => {
+    let isInventoryDisplayActive;
+    cy.getCookie('cxConfigE2E')
+      .should('exist')
+      .then((data) => (isInventoryDisplayActive = data.value.includes('true')));
+
+    const body = xhr.response.body;
+    const code =
+      body.availabilities?.availabilityItems[0]?.unitAvailabilities[0]
+        .productCode;
+    const stock =
+      body.availabilities?.availabilityItems[0]?.unitAvailabilities[0].quantity;
+
+    expect(code).to.equal(productCode);
+
+    cy.get(stockSelector).then(($ele) => {
+      const text = $ele.text().trim();
+
+      if (isInventoryDisplayActive) {
+        // Out of stock
+        if (
+          stock.stockLevelStatus === 'outOfStock' ||
+          functionality === 'outOfStock'
+        ) {
+          expect(text).to.equal(sampleData.stockOutOfStockLabel);
+        } else {
+          if (stock?.stockLevel) {
+            if (
+              stock?.isValueRounded ||
+              functionality === 'categoryThresholdLimitReached'
+            ) {
+              expect(text).to.equal(
+                `${stock.stockLevel}+ ${sampleData.stockLabel}`
+              );
+            } else {
+              expect(text).to.equal(
+                `${stock.stockLevel} ${sampleData.stockLabel}`
+              );
+            }
+          } else {
+            expect(text).to.equal(`${sampleData.stockLabel}`);
+          }
+        }
+      } else {
+        if (
+          stock.stockLevelStatus === 'outOfStock' ||
+          functionality === 'outOfStock'
+        ) {
+          expect(text).to.equal(sampleData.stockOutOfStockLabel);
+        } else {
+          expect(text).to.equal(`${sampleData.stockLabel}`);
+        }
+      }
+    });
+  });
+}
+
+export function testInventoryDisplay(
+  productCode: string,
+  functionality: string = ''
+) {
+  const productDetailsAlias = interceptProductAvailability(productCode);
+  visitProductPage(productCode);
+
+  cy.wait(`@${productDetailsAlias}`)
+    .its('response.statusCode')
+    .should('eq', 200);
+
+  assertInventoryDisplay(productCode, `@${productDetailsAlias}`, functionality);
+}
+
+context(`${'B2C'} - Inventory Display`, () => {
```

Comment:
MINOR: can we replace context with describe as context is only available when invoked via the mocha CLI. And why ${'B2C'}, can we just put 'B2C' ?

Reply:
Done
Although there will be one Inventory display check under which this real time stock display gets triggered.